src/zenml/integrations/aws/step_operators/sagemaker_step_operator.py

```diff
@@ -159,20 +159,36 @@ def launch(
 
         settings = cast(SagemakerStepOperatorSettings, self.get_settings(info))
 
-        session = sagemaker.Session(default_bucket=self.config.bucket)
-        instance_type = settings.instance_type or "ml.m5.large"
+        # Get and default fill SageMaker estimator arguments 
+        estimator_args = settings.estimator_args
+        session = estimator_args.sagemaker_session or sagemaker.Session(default_bucket=self.config.bucket)
+        instance_type = estimator_args.instance_type or "ml.m5.large"
+        instance_count = estimator_args.instance_count or 1
+
+        # Create Estimator
         estimator = sagemaker.estimator.Estimator(
             image_name,
             self.config.role,
             environment=environment,
-            instance_count=1,
+            instance_count=instance_count,
             instance_type=instance_type,
             sagemaker_session=session,
+            **estimator_args
```

Comment:
I'm guessing this would fail if the `estimator_args` also contains `instance_type` as we would then supply two 
values for the same keyword-argument to the function?

Reply:
It does indeed. Estimator initialization now fully relies on the provided `estimator_args`. If `instance_type` is not provided in the dictionary, it is set to `'ml.m5.large'`. This way, the two-value situation should never occur. Same for `environment`, `instance_count`, `sagemaker_session` for which the user dict is overruled by the ZenML step operator.